src/components/Editor/SearchBar.js

```diff
@@ -437,27 +448,41 @@ const SearchBar = React.createClass({
   },
 
   renderSearchTypeToggle() {
-    if (!isEnabled("functionSearch")) {
+    if (!isEnabled("symbolSearch")) {
       return;
     }
+    const { toggleSymbolSearch } = this;
+    const { symbolSearchEnabled, selectedSymbolType } = this.state;
+
+    function searchTypeBtn(searchType) {
+      return dom.button({
+        className: classnames("search-type-btn", {
+          active: symbolSearchEnabled && selectedSymbolType == searchType
+        }),
+        onClick: e => {
+          if (selectedSymbolType == searchType) {
+            toggleSymbolSearch(e, { toggle: true, searchType });
+            return;
+          }
+          toggleSymbolSearch(e, { toggle: false, searchType });
+        }
+      }, searchType);
+    }
 
     return dom.section(
       { className: "search-type-toggles" },
       dom.h1(
         { className: "search-toggle-title" },
         "Search for:"
       ),
-      dom.button({
-        className: classnames("search-type-btn", {
-          active: this.state.functionSearchEnabled
-        }),
-        onClick: e => this.toggleFunctionSearch(e, { toggle: true })
-      }, "functions")
+      searchTypeBtn("functions"),
+      searchTypeBtn("variables"),
+      searchTypeBtn("classes")
```

Comment:
I like that idea but I think we'd need to modify when we call `getSymbols` so that the symbol declarations are available after documents load (getting the declarations is currently done when `doSearch` is called).